### PR TITLE
Adding support to override parameters in serviceFabricDeploy Task.

### PR DESCRIPTION
**Pull request by hiyadav**

### CODE REVIEW

File: Tasks/ServiceFabricDeploy/utilities.ps1 [hunk 1]
```diff
@@ -110,6 +110,41 @@ function Read-PublishProfile
...
+    Param (
+        [String]
+        $ApplicationManifestPath
+    )
+
+    $OverrideParameters = @{}
+    $ApplicationManifestXml = [Xml] (Get-Content -LiteralPath $ApplicationManifestPath)
+    foreach ($param in $ApplicationManifestXml.ApplicationManifest.Parameters.Parameter)
+    {
+        $paramName = $param.Name -replace "\.", '_'
```
Comment (kuvinodms) [inline Tasks/ServiceFabricDeploy/utilities.ps1]: Use this: 
$param.Name -replace "\.", '_' -replace " ", '_'

File: Tasks/ServiceFabricDeploy/utilities.ps1 [hunk 2]
```diff
@@ -110,6 +110,41 @@ function Read-PublishProfile
...
+}
+
+function Get-ApplicationManifesetPath
+{
+    Param (
+        [String]
+        $ApplicationPackagePath
+    )
+
+    $appManifestName = "ApplicationManifest.xml"
```
Comment (kuvinodms) [inline Tasks/ServiceFabricDeploy/utilities.ps1]: Its always better to assign a variable or constant (with appropriate name) to such file name at the top of the function/file

File: Tasks/ServiceFabricDeploy/utilities.ps1 [hunk 3]
```diff
@@ -110,6 +110,41 @@ function Read-PublishProfile
...
 }
 
+function Get-OverridenApplicationParameters
+{
+    Param (
+        [String]
+        $ApplicationManifestPath
+    )
+
+    $OverrideParameters = @{}
```
Comment (sachinma) [inline Tasks/ServiceFabricDeploy/utilities.ps1]: Start the local variables with lowercase

File: Tasks/ServiceFabricDeploy/task.json [hunk 4]
```diff
@@ -74,6 +74,14 @@
             "required": false,
             "helpMarkDown": "Path to the application parameters file. [Variables](https://go.microsoft.com/fwlink/?LinkID=550988) and wildcards can be used in the path. If specified, this will override the value in the publish profile."
         },
+        {
+            "name": "overrideApplicationParameter",
+            "type": "boolean",
+            "label": "Override Application Parameters",
+            "defaultValue": "false",
+            "required": false,
+            "helpMarkDown": "Variables defined in the Build or Release Definition will be matched against the 'key' or 'name' entries in the application manifest file. \n Example: If your application has a parameter defined as below. \n <Parameter Name=\"SampleApp_PartitionCount\" Value=\"1\" /> \n and you want to change the partition count to 2, you can define a release definition or an environment variable \"SampleApp_PartitionCount\" and its value as \"2\". \n\n Note: If same variables are defined in the Release Definition and in the Environment, then the Environment variables will supersede the Release Definition variables."
```
Comment (sachinma) [inline Tasks/ServiceFabricDeploy/task.json]: Is the key and value present in the manifest file? Can you work with Pulkit to get the right message.

**Pull request merged**